### PR TITLE
Remove type attribute

### LINKED ISSUE #162: remove type attribute in fittingproblem classes
In BaseFittingProblem currently have attribute self._type = None. 

In this issue remove this type and instead figure out the type from the child name. The issue is that currently the type is in effect represented twice, which is not needed.

### PR DESCRIPTION
**Pull request by AtomAnu**
#### Description of Work

Fixes #162 

The type name is now obtained by the child name (file name).

#### Testing Instructions

1. Run `example_runScripts.py` and investigate the correctness of the output

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,316 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ChangeListManager">
```
Comment (Anders-Markvardsen) [inline .idea/workspace.xml]: Please remove this file

**Comment by Anders-Markvardsen** [comment]
Hi @AtomAnu, could you alternatively look into using https://stackoverflow.com/questions/11408148/how-to-get-derived-class-name-from-base-class and see if you can use this to remove _type attribute entirely from the base class

File: fitbenchmarking/parsing/parse_fitbenchmark_data.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def __init__(self, fname):
 
         self._name = entries['name']
         self._equation = entries['function']
-        self._type = "FitBenchmark"
+        self._type = ((os.path.basename(__file__)).split('_')[1]).upper()
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/parsing/parse_fitbenchmark_data.py]: I think you can remove self._type from here and in equivalent file for NIST by using suggestion here for getting module name in https://stackoverflow.com/questions/2020014/get-fully-qualified-class-name-of-an-object-in-python in func_def.py

Comment (AtomAnu) [reply fitbenchmarking/parsing/parse_fitbenchmark_data.py]: I will look into that.

**Comment by AtomAnu** [comment]
@Anders-Markvardsen 
I have finished solving this issue.

**Comment by Anders-Markvardsen** [comment]
Could you check the build - looks like some tests are not passing

**Pull request merged**